pkg/util/detector/detector.go

```diff
@@ -528,15 +552,62 @@ func (d *ResourceDetector) BuildClusterResourceBinding(object *unstructured.Unst
 		},
 		Spec: workv1alpha1.ResourceBindingSpec{
 			Resource: workv1alpha1.ObjectReference{
-				APIVersion:      object.GetAPIVersion(),
-				Kind:            object.GetKind(),
-				Name:            object.GetName(),
-				ResourceVersion: object.GetResourceVersion(),
+				APIVersion:                  object.GetAPIVersion(),
+				Kind:                        object.GetKind(),
+				Name:                        object.GetName(),
+				ResourceVersion:             object.GetResourceVersion(),
+				ReplicaResourceRequirements: replicaResourceRequirements,
+				Replicas:                    replicas,
 			},
 		},
 	}
 
-	return binding
+	return binding, nil
+}
+
+// GetObjectResource get the replicas and resources of a object
+func (d *ResourceDetector) GetObjectResource(object *unstructured.Unstructured) (corev1.ResourceList, int32, error) {
+	if object.GetKind() == util.DeploymentKind {
+		replicas, ok, err := unstructured.NestedInt64(object.Object, util.SpecField, util.ReplicasField)
+		if !ok || err != nil {
+			return nil, 0, err
+		}
+		podTemplate, ok, err := unstructured.NestedMap(object.Object, util.SpecField, util.TemplateField)
+		if !ok || err != nil {
+			return nil, 0, err
+		}
+		replicaResourceRequirements, err := d.getReplicaResourceRequirements(podTemplate)
+		if err != nil {
+			return nil, 0, err
+		}
+		return replicaResourceRequirements, int32(replicas), nil
+	}
+	return nil, 0, nil
+}
+
+func (d *ResourceDetector) getReplicaResourceRequirements(object map[string]interface{}) (corev1.ResourceList, error) {
+	var podTemplateSpec *corev1.PodTemplateSpec
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(object, &podTemplateSpec)
+	if err != nil {
+		return nil, err
+	}
+	resPtr := d.calculateResource(&podTemplateSpec.Spec)
+	replicaResourceRequirements := corev1.ResourceList{
+		corev1.ResourceCPU:    *resource.NewMilliQuantity(resPtr.MilliCPU, resource.DecimalSI),
+		corev1.ResourceMemory: *resource.NewQuantity(resPtr.Memory, resource.BinarySI),
+	}
```

Comment:
Why not use `resPtr.ResourceList()` here?

Reply:
ok